ui/lib/kv/addon/components/page/secret/metadata/version-diff.hbs

```diff
@@ -1,11 +1,90 @@
-<KvPageHeader @breadcrumbs={{@breadcrumbs}} @model={{@model}} @pageTitle={{@model.name}}>
+<KvPageHeader @breadcrumbs={{@breadcrumbs}} @model={{@model}} @pageTitle={{@path}}>
   <:tabLinks>
     <LinkTo @route="secret.details" data-test-secrets-tab="Secret">Secret</LinkTo>
     <LinkTo @route="secret.metadata.index" data-test-secrets-tab="Metadata">Metadata</LinkTo>
-    <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Metadata">Version History</LinkTo>
-    <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Metadata">Version Diff</LinkTo>
+    <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Version History">Version History</LinkTo>
+    <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Version Diff">Version Diff</LinkTo>
   </:tabLinks>
 
+  <:toolbarFilters>
+    {{! left side version selector }}
+    <BasicDropdown @class="popup-menu" @horizontalPosition="auto-right" @verticalPosition="below" as |D|>
+      <D.Trigger data-test-version-dropdown class="toolbar-link {{if D.isOpen ' is-active'}}">
+        Version
+        {{this.leftSideVersion}}
+        <Chevron @direction="down" @isButton={{true}} />
+      </D.Trigger>
+      <D.Content @defaultClass="popup-menu-content">
+        <nav class="box menu" aria-label="version-left">
+          <ul class="menu-list">
+            {{#each @metadata.sortedVersions as |versionData|}}
+              <li>
+                <button
+                  type="button"
+                  class="link"
+                  {{on "click" (fn this.selectVersion versionData.version D.actions "left")}}
+                  disabled={{(or versionData.destroyed versionData.deletion_time)}}
+                >
+                  Version
+                  {{versionData.version}}
+                  {{#if (eq versionData.version this.leftSideVersion)}}
+                    <Icon @name="check-circle" class="has-text-success is-pulled-right" />
+                  {{else if versionData.destroyed}}
+                    <Icon @name="x-square-fill" class="has-text-danger is-pulled-right" />
+                  {{else if versionData.deletion_time}}
+                    <Icon @name="x-square-fill" class="has-text-grey is-pulled-right" />
+                  {{/if}}
+                </button>
+              </li>
+            {{/each}}
+          </ul>
+        </nav>
+      </D.Content>
+    </BasicDropdown>
+    {{! right side version selector }}
+    <BasicDropdown @class="popup-menu" @horizontalPosition="auto-right" @verticalPosition="below" as |D|>
```

Comment:
Wondering if there's a way to pass in an `on click` function and make `KvVersionDropdown reusable`

Reply:
As discussed, a nice clean up option. Limiting how much time we're spending on this view.